static/app/views/performance/newTraceDetails/traceModels/traceTree.ssr.spec.tsx

```diff
@@ -91,4 +91,24 @@ describe('server side rendering', () => {
 
     expect(tree.build().serialize()).toMatchSnapshot();
   });
+
+  it('does not reparent if server handler does not have SSR reparent reason', () => {
+    const tree = TraceTree.FromTrace(ssrTrace, traceMetadata);
+
+    // The automatic pageload reparenting should have happened
+    const pageload = tree.root.children[0]!.children[0]!;
+    const serverHandler = pageload.children[0]!;
```

Comment:
Wdyt about asserting that both of these are actually pageload and server handler nodes?
You could write it as
```tsx
const pageload = tree.root.find(p => p.op === 'pageload');
const serverHandler = tree.root.find(p => p.op === 'http.server');
expect(pageload.parent).toBe(serverHandler);
```